server/src/main/java/org/opensearch/search/profile/Profilers.java

```diff
@@ -64,7 +66,9 @@ public Profilers(ContextIndexSearcher searcher, boolean isConcurrentSegmentSearc
 
     /** Switch to a new profile. */
     public QueryProfiler addQueryProfiler() {
-        QueryProfiler profiler = new QueryProfiler(isConcurrentSegmentSearchEnabled);
+        QueryProfiler profiler = isConcurrentSegmentSearchEnabled
+            ? new ConcurrentQueryProfiler()
+            : new QueryProfiler(new InternalQueryProfileTree());
```

Comment:
We should instantiate `ConcurrentQueryProfiler` as `new ConcurrentQueryProfiler(new ConcurrentQueryProfileTree)` to be consistent with `QueryProfiler`

Reply:
It has updated now. Thank you.